idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -504,6 +548,11 @@ def run_host_filtering(fastq_files, initial_file_type_for_log, lazy_run, stats):
                       after_filename=os.path.join(RESULT_DIR, STAR_OUT1),
                       after_filetype=initial_file_type_for_log)
 
+    # start downloading gsnap genome after STAR step
+    local_gsnap_genome = fetch_from_s3(GSNAP_GENOME,
+                                       os.path.join(REF_DIR, os.path.basename(GSNAP_GENOME)),
+                                       auto_unzip=False)
```

Comment:
i think this wasn't using s3mi because you didn't tell it to allow_s3mi=True

but in any case you shouldn't just copy a tar file from s3 (by whatever means) and then untar it, because that just doubles the amount of data that needs to be written to disk, which is the actual slow part

May I suggest you simply use the command `s3mi cat <s3_path> | tar xvf - -C /mnt/idseq/ref` or if you don't like s3mi just do `aws s3 cp <s3_path> - | tar xvf - -C /mnt/idseq/ref` and start that in a thread at the beginning.   By the time its result is needed, it would have finished.

Reply:
OK. I was confused about how fetch_from_s3 work